docs/source/examples/qlora_finetune.rst

```diff
@@ -0,0 +1,285 @@
+
+.. note::
+    Make sure to correctly point to the location of your Llama2 weights and tokenizer. This can be done
+    either by adding :code:`checkpointer.checkpoint_files=[my_model_checkpoint_path] tokenizer_checkpoint=my_tokenizer_checkpoint_path`
+    or by directly modifying the :code:`7B_qlora_single_device.yaml` file. See our :ref:`config_tutorial_label`
+    for more details on how you can easily clone and modify TorchTune configs.
+
+By default, this run should log peak memory stats at model initialization time and every 100
+iterations during training. Let's understand the memory savings enabled by QLoRA on top of LoRA training. LoRA training
+can be run as follows:
+
+.. code-block:: bash
+
+    tune run lora_finetune_single_device --config recipes/configs/llama2/7B_lora_single_device.yaml
+
+You should see the memory usage printed out during model initialization:
+
+.. code-block:: python
+
+  Memory Stats after model init::
+  GPU peak memory allocation: 13.96 GB
+  GPU peak memory reserved: 13.98 GB
+  GPU peak memory active: 13.96 GB
+
+
+As well as during training:
+
+.. code-block:: python
+
+  Memory Stats::
+  GPU peak memory allocation: 14.40 GB
+  GPU peak memory reserved: 15.57 GB
+  GPU peak memory active: 14.40 GB
+
+
+Comparing to the memory usage during model initialization for QLoRA, we see about a 35% decrease in peak memory reserved:
+
+.. code-block:: python
+
+  Memory Stats after model init::
+  GPU peak memory allocation: 7.36 GB
+  GPU peak memory reserved: 9.13 GB
+  GPU peak memory active: 7.36 GB
+
+As well as a 40% decrease in peak memory reserved during training:
+
+.. code-block:: python
+
+  Memory Stats::
+  GPU peak memory allocation: 5.54 GB
+  GPU peak memory reserved: 9.29 GB
+  GPU peak memory active: 5.54 GB
+
+From the logs, one can see that the out-of-the-box training performance is quite slow, slower than 1 iteration per
+second:
+
+.. code-block:: python
+
+  1|149|Loss: 0.9157477021217346:   1%|          | 149/25880 [02:08<6:14:19,  1.15it/s
+
+To speed things up, we can leverage ``torch.compile`` to compile our model and run the compiled result. To work with
+QLoRA training, a nightly build of PyTorch must be used. To update PyTorch to the latest nightly,
+please see `the installation instructions <https://pytorch.org/get-started/locally/>`_. Once updated,
+users can specify the compile flag as ``True`` via a config override:
+
+.. code-block:: bash
+
+    tune run lora_finetune_single_device --config recipes/configs/llama2/7B_lora_single_device.yaml compile=True
+
+From the logs, we can see about a 200% speed up (after a few hundred iterations once the training has stabilized):
+
+.. code-block:: python
+
+  1|228|Loss: 0.8158286809921265:   1%|          | 228/25880 [11:59<1:48:16,  3.95it/s
+
+A comparison of the smoothed loss curves between QLoRA and LoRA can be seen below (purple being the QLoRA loss curve).
+
+.. image:: /_static/img/qlora_experiment.png
```

Comment:
Btw you can also add explicit labels to the two lines in the figure, e.g. iconic-pasma-57 -> LoRA and azure-bird-56 -> QLoRA

Reply:
Sorry to be annoying, but can you filter the x-axis to [0, 1000] or something in wandb and reupload? Otherwise it looks weird that one is running longer.